pkg/kubelet/network/dns/dns.go

```diff
@@ -98,17 +104,35 @@ func (c *Configurer) getClusterDNS(pod *v1.Pod) ([]net.IP, error) {
 		networkName = arktosv1.NetworkDefault
 	}
 
-	if network, err := c.arktosV1Client.NetworksWithMultiTenancy(pod.Tenant).Get(networkName, metav1.GetOptions{}); err == nil {
-		if network.Status.Phase != arktosv1.NetworkReady {
-			return nil, fmt.Errorf("network %s/%s is in %q phase, not ready", network.Tenant, network.Name, network.Status.Phase)
+	if c.serviceLister == nil {
+		return nil, fmt.Errorf("failed to get serviceLister when getting service IP for network")
+	}
+	services, err := c.serviceLister.List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("failed to list services when getting service IP for network")
+	}
+
+	const kubeDNSPrefix = "kube-dns-"
+	var kubeDNSServiceName = kubeDNSPrefix + networkName
+
+	// Get all services and find out the cluster IP of service kube-dns-{network}
+	for i := range services {
```

Comment:
consider speed up the search, if it is possible.

Reply:
To do optimization: 
For 1 TP: consider use {TENANT}+{NAMESPACE}+{SERVICE NAME} as key to create serviceMAP, then use serviceMAP to find out corresponding service IP of kube-dns-{network}

For multi-TP:  have to consider how to identify serviceLister from different TPs.